<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>exposure &amp; epigenome | Chiara Herzog</title> <meta name="author" content="Chiara Herzog"> <meta name="description" content="assessing impacts of the environment (exposome) on dna methylation patterns"> <meta name="keywords" content="chiara herzog, methylation, biomarkers of aging, prevention research, cancer detection"> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.21.4/dist/bootstrap-table.min.css"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.4.0/css/all.min.css" integrity="sha256-HtsXJanqjKTc8vVQjO4YMhiqFoXkfBsjBWcX91T1jr8=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%8C%99&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://chiaraherzog.github.io/projects/ongoing/exposure_epigenome/"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js?96d6b3e1c3604aca8b6134c7afdd5db6"></script> <script src="/assets/js/dark_mode.js?9b17307bb950ffa2e34be0227f53558f"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"><span class="font-weight-bold">Chiara </span>Herzog</a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about</a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv</a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">projects</a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications</a> </li> <li class="nav-item "> <a class="nav-link" href="/tools/">tools</a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">teaching</a> </li> <li class="nav-item "> <a class="nav-link" href="/outreach/">media &amp; outreach</a> </li> <li class="nav-item"> <a class="nav-link" href="mailto:%63%68%69%61%72%61.%68%65%72%7A%6F%67@%6B%63%6C.%61%63.%75%6B" title="email"> <i class="fas fa-envelope"></i> </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fas fa-moon"></i> <i class="fas fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5"> <div class="post"> <header class="post-header"> <h1 class="post-title">exposure &amp; epigenome</h1> <p class="post-description">assessing impacts of the environment (exposome) on dna methylation patterns</p> </header> <article> <p>I am investigating the impact of various environmental factors such as age, smoking, diet, or viral infections, on DNA methylation patterns. Part of this work is undertaken with the Human Exposome Assessment Platform (<a href="https://heap-exposome.eu/work-packages/" target="_blank" rel="external nofollow noopener">HEAP</a>) Horizon 2020 programme.</p> <p>Publications include:</p> <div class="publications"> <h2 class="bibliography">2024</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/smoking-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/smoking-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/smoking-1400.webp"></source> <img src="/assets/img/publication_preview/smoking.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="smoking.png" data-zoomable="" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="canresvaping" class="col-sm-8"> <div class="title">Cigarette smoking and e-cigarette use induce shared DNA methylation changes linked to carcinogenesis</div> <div class="author"> <em>Chiara Herzog</em>, Allison Jones, Iona Evans, and <span class="more-authors" title="click to view 7 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '7 more authors' ? 'Janhavi R. Raut, Michal Zikan, David Cibula, Andrew Wong, Hermann Brenner, Rebecca C. Richmond, Martin Widschwendter' : '7 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '0.01'); ">7 more authors</span> </div> <div class="periodical"> <em>Cancer Research</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1158/0008-5472.can-23-2957" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1158/0008-5472.can-23-2957"></span> <span class="__dimensions_badge_embed__" data-doi="10.1158/0008-5472.can-23-2957" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Tobacco use is a major modifiable risk factor for adverse health outcomes, including cancer, and elicits profound epigenetic changes thought to be associated with long-term cancer risk. While electronic cigarettes (e-cigarettes) have been advocated as harm reduction alternatives to tobacco products, recent studies have revealed potential detrimental effects, highlighting the urgent need for further research into the molecular and health impacts of e-cigarettes. Here, we applied computational deconvolution methods to dissect the cell- and tissue-specific epigenetic effects of tobacco or e-cigarette use on DNA methylation (DNAme) in over 3,500 buccal/saliva, cervical, or blood samples, spanning epithelial and immune cells at directly and indirectly exposed sites. The 535 identified smoking-related DNAme loci (CpGs) clustered into four functional groups, including detoxification or growth signaling, based on cell type and anatomical site. Loci hypermethylated in buccal epithelial cells of smokers associated with NOTCH1/RUNX3/growth factor receptor signaling also exhibited elevated methylation in cancer tissue and progressing lung carcinoma in situ lesions, and hypermethylation of these sites predicted lung cancer development in buccal samples collected from smokers up to 22 years prior to diagnosis, suggesting a potential role in driving carcinogenesis. Alarmingly, these CpGs were also hypermethylated in e-cigarette users with a limited smoking history. This study sheds light on the cell type-specific changes to the epigenetic landscape induced by smoking-related products.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/widhpv-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/widhpv-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/widhpv-1400.webp"></source> <img src="/assets/img/publication_preview/widhpv.jpg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="widhpv.jpg" data-zoomable="" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="widhpv" class="col-sm-8"> <div class="title">HPV‐induced host epigenetic reprogramming is lost upon progression to high‐grade cervical intraepithelial neoplasia</div> <div class="author"> <em>Chiara Herzog*</em>, Charlotte D. Vavourakis*, James E. Barrett, and <span class="more-authors" title="click to view 6 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '6 more authors' ? 'Gerlinde Karbon, Andreas Villunger, Jiangrong Wang, Karin Sundström, Joakim Dillner, Martin Widschwendter' : '6 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '0.01'); ">6 more authors</span> </div> <div class="periodical"> <em>International Journal of Cancer</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://onlinelibrary.wiley.com/doi/10.1002/ijc.34477" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1002/ijc.34477"></span> <span class="__dimensions_badge_embed__" data-doi="10.1002/ijc.34477" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>The impact of a pathogen on host disease can only be studied in samples covering the entire spectrum of pathogenesis. Persistent oncogenic Human Papilloma Virus (HPV) infection is the most common cause for cervical cancer. Here, we investigate HPV‐induced host epigenome‐wide changes prior to development of cytological abnormalities. Using cervical sample methylation array data from disease‐free women with or without an oncogenic HPV infection, we develop the WID (Women’s cancer risk identification)‐HPV, a signature reflective of changes in the healthy host epigenome related to high‐risk HPV strains (AUC=0.78, 95% CI: 0.72‐0.85, in non‐diseased women). Looking at HPV‐associated changes across disease development, HPV‐infected women with minor cytological alterations (cervical intraepithelial neoplasia grade 1/2, CIN1/2), but surprisingly not those with precancerous changes or invasive cervical cancer (CIN3+), show an increased WID‐HPV index, indicating the WID‐HPV may reflect a successful viral clearance response absent in progression to cancer. Further investigation revealed the WID‐HPV is positively associated with apoptosis (ρ=0.48; p&lt;0.001) and negatively associated with epigenetic replicative age (ρ=‐0.43; p&lt;0.001). Taken together, our data suggest the WID‐HPV captures a clearance response associated with apoptosis of HPV‐infected cells. This response may be dampened or lost with increased underlying replicative age of infected cells, resulting in progression to cancer. This article is protected by copyright. All rights reserved.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/qtbc-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/qtbc-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/qtbc-1400.webp"></source> <img src="/assets/img/publication_preview/qtbc.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="qtbc.png" data-zoomable="" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="qtbc" class="col-sm-8"> <div class="title">DNA methylation at quantitative trait loci (mQTLs) varies with cell type and nonheritable factors and may improve breast cancer risk assessment</div> <div class="author"> <em>Chiara Herzog</em>, Allison Jones, Iona Evans, and <span class="more-authors" title="click to view 8 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '8 more authors' ? 'Michal Zikan, David Cibula, Nadia Harbeck, Nicoletta Colombo, Angelique Flöter Rådestad, Kristina Gemzell-Danielsson, Nora Pashayan, Martin Widschwendter' : '8 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '0.01'); ">8 more authors</span> </div> <div class="periodical"> <em>npj Precision Oncology</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41698-023-00452-2" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="/assets/pdf/qtBC_manuscript.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1038/s41698-023-00452-2"></span> <span class="__dimensions_badge_embed__" data-doi="10.1038/s41698-023-00452-2" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>To individualise breast cancer (BC) prevention, markers to follow a person’s changing environment and health extending beyond static genetic risk scores are required. Here, we analysed cervical and breast DNA methylation (n=1848) and single nucleotide polymorphisms (n=1442) and demonstrate that a linear combination of methylation levels at 104 BC-associated methylation quantitative trait loci (mQTL) CpGs, termed the WID™-qtBC index, can identify women with breast cancer in hormone-sensitive tissues (AUC=0.71 [95% CI: 0.65–0.77] in cervical samples). Women in the highest combined risk group (high polygenic risk score and WID™-qtBC) had a 9.6-fold increased risk for BC [95% CI: 4.7–21] compared to the low-risk group and tended to present at more advanced stages. Importantly, the WID™-qtBC is influenced by non-genetic BC risk factors, including age and body mass index, and can be modified by a preventive pharmacological intervention, indicating an interaction between genome and environment recorded at the level of the epigenome. Our findings indicate that methylation levels at mQTLs in relevant surrogate tissues could enable integration of heritable and non-heritable factors for improved disease risk stratification.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/clocks-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/clocks-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/clocks-1400.webp"></source> <img src="/assets/img/publication_preview/clocks.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="clocks.png" data-zoomable="" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="widrea" class="col-sm-8"> <div class="title">Susceptibility to hormone-mediated cancer is reflected by different tick rates of the epithelial and general epigenetic clock</div> <div class="author"> James E. Barrett*, <em>Chiara Herzog*</em>, Yoo-Na Kim*, and <span class="more-authors" title="click to view 12 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '12 more authors' ? 'Thomas E. Bartlett, Allison Jones, Iona Evans, David Cibula, Michal Zikan, Line Bjørge, Nadia Harbeck, Nicoletta Colombo, Sacha J. Howell, Angelique Flöter Rådestad, Kristina Gemzell-Danielsson, Martin Widschwendter' : '12 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '0.01'); ">12 more authors</span> </div> <div class="periodical"> <em>Genome Biology</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1186/s13059-022-02603-3" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1186/s13059-022-02603-3"></span> <span class="__dimensions_badge_embed__" data-doi="10.1186/s13059-022-02603-3" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>A variety of epigenetic clocks utilizing DNA methylation changes have been developed; these clocks are either tissue-independent or designed to predict chronological age based on blood or saliva samples. Whether discordant tick rates between tissue-specific and general epigenetic clocks play a role in health and disease has not yet been explored. Here we analyze 1941 cervical cytology samples, which contain a mixture of hormone-sensitive cervical epithelial cells and immune cells, and develop the WID general clock (Women’s IDentification of risk), an epigenetic clock that is shared by epithelial and immune cells and optimized for cervical samples. We then develop the WID epithelial clock and WID immune clock, which define epithelial- and immune-specific clocks, respectively. We find that the WID-relative-epithelial-age (WID-REA), defined as the difference between the epithelial and general clocks, is significantly reduced in cervical samples from pre-menopausal women with breast cancer (OR 2.7, 95% CI 1.28-5.72). We find the same effect in normal breast tissue samples from pre-menopausal women at high risk of breast cancer and show that potential risk reducing anti-progesterone drugs can reverse this. In post-menopausal women, this directionality is reversed. Hormone replacement therapy consistently leads to a significantly lower WID-REA in cancer-free women, but not in post-menopausal women with breast or ovarian cancer. Our findings imply that there are multiple epigenetic clocks, many of which are tissue-specific, and that the differential tick rate between these clocks may be an informative surrogate measure of disease risk.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"><li> <div class="row"> <div class="col-sm-2 preview"> <figure> <picture> <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/senolysis_review-480.webp"></source> <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/senolysis_review-800.webp"></source> <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/senolysis_review-1400.webp"></source> <img src="/assets/img/publication_preview/senolysis_review.jpg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="senolysis_review.jpg" data-zoomable="" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="senolysisreview" class="col-sm-8"> <div class="title">Dietary restriction in senolysis and prevention and treatment of disease</div> <div class="author"> Sepideh Aminzadeh-Gohari, Barbara Kofler, and <em>Chiara Herzog</em> </div> <div class="periodical"> <em>Critical Reviews in Food Science and Nutrition</em>, 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1080/10408398.2022.2153355" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="badges"> <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1080/10408398.2022.2153355"></span> <span class="__dimensions_badge_embed__" data-doi="10.1080/10408398.2022.2153355" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span> </div> <div class="abstract hidden"> <p>Aging represents a key risk factor for a plethora of diseases. Targeting detrimental processes which occur during aging, especially before onset of age-related disease, could provide drastic improvements in healthspan. There is increasing evidence that dietary restriction (DR), including caloric restriction, fasting, or fasting-mimicking diets, extend both lifespan and healthspan. This has sparked interest in the use of dietary regimens as a non-pharmacological means to slow aging and prevent disease. Here, we review the current evidence on the molecular mechanisms underlying DR-induced health improvements, including removal of senescent cells, metabolic reprogramming, and epigenetic rejuvenation.</p> </div> </div> </div> </li></ol> </div> </article> </div> </div> <footer class="fixed-bottom"> <div class="container mt-0"> © Copyright 2025 Chiara Herzog. Last updated: April 02, 2025. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js"></script> <script defer src="https://unpkg.com/bootstrap-table@1.21.4/dist/bootstrap-table.min.js"></script> <script src="/assets/js/no_defer.js?d633890033921b33e0ceb13d22340a9c"></script> <script defer src="/assets/js/common.js?acdb9690d7641b2f8d40529018c71a01"></script> <script defer src="/assets/js/copy_code.js?c9d9dd48933de3831b3ee5ec9c209cac" type="text/javascript"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script async src="https://www.googletagmanager.com/gtag/js?id=G-6ZLF3RJPVS"></script> <script>function gtag(){window.dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-6ZLF3RJPVS");</script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>